File: app/src/users/routers/users_actions.js
const express = require("express");
const router = express.Router();
const { TourismItem } = require("../tourism/schemas/tourism"); // Import the TourismItem model or class from your tourism schema file
const userActionsRepo = require("../users/repository/user_actions_repo"); // Import your user_actions_repo or repository functions

// POST create user request
router.post("/request", async (req, res) => {
    const { username } = req.body; // Extract username from the request body
    const request = req.body.request; // Extract the request object from the request body

    try {
        // Implement the logic to create a user request here
        const result = await userActionsRepo.createUserRequest(
            username,
            request
        );
        res.status(201).json(result);
    } catch (error) {
        res.status(500).json({ error: "Internal server error" });
    }
});

module.exports = router;

File: app/src/users/routers/users_actions.py
# from app.src.tourism.schemas.tourism import TourismItem
# from app.src.users.repository import user_actions_repo
# from fastapi import APIRouter, HTTPException, status

# router = APIRouter(
#     prefix="/users/action",
#     tags=["Users"],
# )


# @router.post("/request")
# async def create_user_request(username: str, request: TourismItem) -> dict:
#     """Create user request"""
#     try:
#         return user_actions_repo.create_user_request(username, request)
#     except Exception as e:
#         raise HTTPException(
#             status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e)
#         )
